rlkit/core/online_rl_algorithm.py

```diff
@@ -0,0 +1,199 @@
+from collections import OrderedDict
+
+import gtimer as gt
+
+from rlkit.core import logger, eval_util
+from rlkit.data_management.replay_buffer import ReplayBuffer
+from rlkit.samplers.data_collector import PathCollector
+
+
+def _get_epoch_timings(epoch):
+    times_itrs = gt.get_times().stamps.itrs
+    times = OrderedDict()
+    epoch_time = 0
+    for key in sorted(times_itrs):
+        time = times_itrs[key][-1]
+        epoch_time += time
+        times['time/{} (s)'.format(key)] = time
+    times['time/epoch (s)'] = epoch_time
+    times['time/total (s)'] = gt.get_times().total
+    return times
+
+
+class OnlineRLAlgorithm(object):
+    def __init__(
+            self,
+            trainer,
+            exploration_env,
+            evaluation_env,
+            exploration_data_collector: PathCollector,
+            evaluation_data_collector: PathCollector,
```

Comment:
Also, I think `evaluation_data_collector` **should** stay as a `PathCollector`. This will be useful if (e.g.) we later add an asynchronous `PathCollector`.